src/main/io/osd.c

```diff
@@ -325,9 +325,37 @@ static bool osdDrawSingleElement(uint8_t item)
 
             if (FLIGHT_MODE(PASSTHRU_MODE))
                 p = "PASS";
-            else if (FLIGHT_MODE(FAILSAFE_MODE))
+            else if (FLIGHT_MODE(FAILSAFE_MODE)) {
                 p = "!FS!";
-            else if (FLIGHT_MODE(HEADFREE_MODE))
+                // See failsafe.h for each phase explanation
+                switch (failsafePhase()) {
+                    case FAILSAFE_RX_LOSS_IDLE:
+                        p = "F:RX"; // RX to indicate that failsafe is waiting
+                                   // for the RX to become available
+                        break;
+#ifdef NAV
+                    case FAILSAFE_RETURN_TO_HOME:
+                        p = "F:RT"; // RT for RETURN
+                        break;
+#endif
+                    case FAILSAFE_LANDING:
+                        p = "F:LA"; // LA for LANDING
+                        break;
+                    case FAILSAFE_LANDED:
+                        p = "F:DR"; // DR for DROP
+                        break;
+                    case FAILSAFE_RX_LOSS_MONITORING:
+                        p = "F:WA"; // WA for WAITING
+                        break;
+                    case FAILSAFE_IDLE:
+                    case FAILSAFE_RX_LOSS_DETECTED:
+                    case FAILSAFE_RX_LOSS_RECOVERED:
+                        /* Failsafe not active or phase is very brief.
```

Comment:
@digitalentity I think you got it wrong. It doesn't emit a warning right now because the switch is handling all possible values of the enum without using a `default`. If someone adds a a new value in `enum failsafePhase_e`, then the compiler will emit a warning because the new value won't be handled by that switch.

This ensures that whenever a new value is added, the one adding the value gets a warning indicating that she should update `io/osd.c` too. Otherwise it's too easy to miss. If the switch had a `default` case, any new values added will be handled by the `default` case, avoiding the very desirable warning.

Totally agree with `-Werror` thought, in fact I usually run `CFLAGS=-Werror make...`